src/main/java/net/fabricmc/loom/configuration/providers/mappings/IntermediaryMappingsProvider.java

```diff
@@ -45,24 +55,59 @@ public abstract class IntermediaryMappingsProvider extends IntermediateMappingsP
 	public abstract Property<Boolean> getRefreshDeps();
 
 	@Override
-	public void provide(Path tinyMappings) throws IOException {
+	public void provide(Path tinyMappings, Project project) throws IOException {
 		if (Files.exists(tinyMappings) && !getRefreshDeps().get()) {
 			return;
 		}
 
 		// Download and extract intermediary
 		final Path intermediaryJarPath = Files.createTempFile(getName(), ".jar");
 		final String encodedMcVersion = UrlEscapers.urlFragmentEscaper().escape(getMinecraftVersion().get());
-		final String url = getIntermediaryUrl().get().formatted(encodedMcVersion);
+		final String urlRaw = getIntermediaryUrl().get();
 
-		LOGGER.info("Downloading intermediary from {}", url);
+		if (project != null && urlRaw.equals(LoomGradleExtensionApiImpl.DEFAULT_INTERMEDIARY_URL)) {
+			final Configuration config = project.getConfigurations()
+					.getByName(Constants.Configurations.INTERMEDIARY_MAPPINGS);
+			try {
+				config.defaultDependencies(new Action<DependencySet>() {
+					@Override
+					public void execute(final DependencySet dependencySet) {
+						final ModuleDependency intermediaryDep = (ModuleDependency) project.getDependencies()
+								.create("net.fabricmc:intermediary:" + encodedMcVersion);
+						intermediaryDep.artifact(new Action<DependencyArtifact>() {
```

Comment:
Would [DependencyFactory::create](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/dsl/DependencyFactory.html#create-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-) make it easier to specify the classifier? I havent used it so im not 100% sure.